src/System.Runtime.Numerics/tests/BigInteger/ctor.cs

```diff
@@ -924,7 +924,7 @@ public static void RunCtorByteArrayTests()
             // ctor(byte[]): array is large
             for (int i = 0; i < s_samples; i++)
             {
-                tempByteArray = new byte[s_random.Next(16384, 2097152)];
+                tempByteArray = new byte[s_random.Next(16384, 262144)];
```

Comment:
Yeah, that's nasty. And I don't know what's the right approach either. I think `stackalloc` is more efficient than `fixed`, since it doesn't lock part of the managed heap? If moving to `fixed` is better than requiring the caller to increase the stack size if needed (millions of bits are already quite huge numbers for my taste), we need to update multiplication too (there was no test for such big arrays).

Who can help here?